test/junit/scala/collection/IteratorTest.scala

```diff
@@ -221,6 +221,14 @@ class IteratorTest {
     assertSameElements(List(3) ++ List(1, 2, 3).reverseIterator.drop(1), List(3, 2, 1))
   }
 
+  @Test def `drop does overflow t13052`: Unit = {
+    var counter = 0L
+    val it = Iterator.continually(counter.tap(_ => counter += 1))
+    val n = 1_000_000_000
+    val dropped = it.drop(n).drop(n).drop(n)
+    assertEquals(3L * n, dropped.next())
+  }
```

Comment:
This is 4-5 secs on my old machine. It would normally go in a "slow test" group.